src/mlio/default_instance_reader.cxx

```diff
@@ -50,7 +53,7 @@ default_instance_reader::default_instance_reader(data_reader_params const &prm,
         throw std::invalid_argument{
             "The shard index must be less than the number of shards."};
     }
-
+    corrupt_split_record_ = false;
```

Comment:
This variable should be initialized either in the constructor initializer list or in the header file using in-place initialization (preferred).